device/iot-device-samples/pnp-device-sample/temperature-controller-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/TemperatureController.java

```diff
@@ -46,9 +49,19 @@
     private static final String THERMOSTAT_2 = "thermostat2";
     private static final String SERIAL_NO = "SR-123456";
 
+    // Environmental variables for Dps
+    private static final String deviceSecurityType = System.getenv("IOTHUB_DEVICE_SECURITY_TYPE");
+    private static final String scopeId = System.getenv("IOTHUB_DEVICE_DPS_ID_SCOPE");
+    private static final String globalEndpoint = System.getenv("IOTHUB_DEVICE_DPS_ENDPOINT");
+    private static final String deviceSymmetricKey = System.getenv("IOTHUB_DEVICE_DPS_DEVICE_KEY");
+    private static final String registrationId = System.getenv("IOTHUB_DEVICE_DPS_DEVICE_ID");
```

Comment:
I've separated the args validation etc logic into a different class, so that the actual sample is not cluttered. How do you feel about doing that here as well?

Reply:
That's a good idea. I'll look into that after this PR gets in.